subprojects/logging/src/main/java/org/gradle/internal/logging/console/DefaultScalableElapsedTimeFormatter.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.logging.console;
+
+public class DefaultScalableElapsedTimeFormatter {
+    public String format(long elapsedTimeMs) {
+        double elapsedTime = elapsedTimeMs / 1000.0;
+
+        if (elapsedTime < 60.0) {
+            return format(elapsedTime, 1, "s");
+        }
+
+        elapsedTime /= 60.0;
+        if (elapsedTime < 60.0) {
+            return format(elapsedTime, 1, "m");
+        }
+
+        elapsedTime /= 60.0;
+        if (elapsedTime < 10.0) {
+            return format(elapsedTime, 2, "h");
+        } else if (elapsedTime < 100.0) {
+            return format(elapsedTime, 1, "h");
+        } else {
+            return format(elapsedTime, 0, "h");
```

Comment:
Missed opportunity for an "easter egg" ;) — just set it to "a really, really long time"